python/stempy/image/__init__.py

```diff
@@ -402,10 +396,17 @@ def com_sparse(electron_counts, frame_dimensions, scan_dimensions, crop_to=None)
                 frame is used as the center of mass.
         :rtype: numpy.ndarray (2D)
         """
-    com = np.zeros((2, scan_dimensions[0] * scan_dimensions[1]), np.float32)
-    for ii, ev in enumerate(electron_counts):
+    com = np.zeros((2, array.num_scans), np.float32)
+    for scan_position in range(array.num_scans):
+        data_indices = array.data_indices(scan_position)
+        if len(data_indices) != 0:
+            # Combine the sparse arrays into one array that we will use
+            ev = np.hstack(array.data[data_indices])
+        else:
+            ev = []
+
         if len(ev) > 0:
-            x, y = np.unravel_index(ev, frame_dimensions)
+            x, y = np.unravel_index(ev, array.frame_shape)
```

Comment:
I have found that `np.unravel_index()` is slower than simply:
```python
x  = ev // frame_shape[0]
y = ev % frame_shape[1]
```
For speed, we might be consider using this instead to avoid the overhead of calling the function and all the extra checks it does.

Reply:
This is now fixed.